service/src/java/org/apache/hive/service/cli/operation/HiveCommandOperation.java

```diff
@@ -110,6 +116,15 @@ public void runInternal() throws HiveSQLException {
       String[] tokens = command.split("\\s");
       String commandArgs = command.substring(tokens[0].length()).trim();
 
+      // For ShowProcesslistProcessor , session and operation level details  are fetched
+      // from SessionManager which can not be accessed from commandProcessor. So, we need to create DTO here..
+      if (commandProcessor instanceof ShowProcesslistProcessor) {
+        List<ProcessListInfo> liveQueries = getLiveQueryInfos(parentSession);
```

Comment:
nit: how about moving the `getLiveQueryInfos(parentSession)` to `ShowProcesslistProcessor`, e.g, change the setup(liveQueries) to `setup(parentSession)`?

Reply:
That was the initial thought but parentSession belong to _hive-service_ module and is not available in hive-exec. We only have SessionState available in ql but it does not have much info about running operations.

Reply:
i don't think HiveCommandOperation is the proper place for `getLiveQueryInfos`. Maybe we could extract `ShowProcesslistProcessor` into the service package and move this logic inside?

Reply:
@deniskuzZ I have extracted the logic in a separate class, `ShowProcesslistOperation` (which extends HiveCommandOperation). Thanks for the input!